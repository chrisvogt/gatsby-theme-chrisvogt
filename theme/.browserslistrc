# Modern browsers that support spread operators and classes natively
# This reduces the need for @babel/plugin-transform-spread and @babel/plugin-transform-classes

# Chrome/Edge (last 2 versions)
last 2 Chrome versions
last 2 Edge versions

# Firefox (last 2 versions)
last 2 Firefox versions

# Safari (last 2 versions)
last 2 Safari versions

# iOS Safari (last 2 versions)
last 2 iOS versions

# Android Chrome (last 2 versions)
last 2 Android versions

# Don't support IE
not IE 11
not IE 10
not IE 9

# Don't support very old versions
> 0.5%
not dead 