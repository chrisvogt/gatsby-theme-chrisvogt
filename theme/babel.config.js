// This is the Babel configuration, which is used by Jest to transpile the theme
// source code when running tests. This is not utilized by GatsbyJS. To use your
// own Babel config with GatsbyJS, please review the following documentation:
// https://www.gatsbyjs.com/docs/babel/
module.exports = {
  presets: ['@babel/preset-env', '@babel/preset-react']
}
