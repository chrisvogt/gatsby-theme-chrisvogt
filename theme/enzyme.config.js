const { configure, mount, render, shallow } = require('enzyme')
const Adapter = require('enzyme-adapter-react-16')

configure({
  adapter: new Adapter()
})

// loader shim
global.___loader = { enqueue: () => {} }

global.mount = mount
global.render = render
global.shallow = shallow

console.error = message => {
  throw new Error(message)
}
