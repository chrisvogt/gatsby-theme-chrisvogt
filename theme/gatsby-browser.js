import 'prismjs/themes/prism-solarizedlight.css'
import 'prismjs/plugins/line-numbers/prism-line-numbers.css'

import './src/styles/global.css'

export { default as wrapRootElement } from './wrapRootElement'

// See https://fossies.org/linux/gatsby/examples/using-reach-skip-nav/README.md
export const onRouteUpdate = ({ location, prevLocation }) => {
  if (prevLocation !== null) {
    const skipLink = document.querySelector('[data-reach-skip-link]') // Ccomes with the <SkipNavLink> component.
    if (skipLink) {
      skipLink.focus()
    }
  }
}
