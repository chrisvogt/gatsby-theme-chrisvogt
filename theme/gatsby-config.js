const path = require('path')

module.exports = () => ({
  siteMetadata: {
    avatarURL:
      'https://res.cloudinary.com/chrisvogt/image/upload/v1573025803/avatar_2x_srlojo.png',
    baseURL: 'https://www.privatesphere.io',
    description:
      'A GatsbyJS blog theme with built-in social widgets for Instagram, Goodreads, GitHub, and Spotify.',
    footerText: 'Made in San Francisco',
    headline: 'Private Sphere',
    imageURL: '',
    languageCode: 'en',
    social: {
      twitterUsername: ''
    },
    subhead: 'A Gatsby blog theme with built-in social widgets',
    title: 'Private Sphere',
    titleTemplate: '%s Â· Private Sphere',
    widgets: {
      github: {
        username: '',
        widgetDataSource: ''
      },
      goodreads: {
        username: '',
        widgetDataSource: ''
      },
      instagram: {
        username: '',
        widgetDataSource: ''
      },
      spotify: {
        username: '',
        widgetDataSource: ''
      },
      steam: {
        username: '',
        widgetDataSource: ''
      }
    }
  },
  plugins: [
    {
      resolve: 'gatsby-plugin-page-creator',
      options: {
        path: path.join(__dirname, 'src/pages')
      }
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        path: path.join(__dirname, 'src/data')
      }
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        path: `content`,
        name: `content`
      }
    },
    `gatsby-plugin-sharp`,
    `gatsby-transformer-sharp`,
    {
      resolve: 'gatsby-plugin-mdx',
      options: {
        gatsbyRemarkPlugins: [
          `gatsby-remark-prismjs`,
          `gatsby-remark-images`,
          `gatsby-remark-copy-linked-files`
        ],
        plugins: [
          // FIX(cvogt): this plugin is defined here as a temporary fix for the bug
          // described in gatsbyjs/gatsby#15486
          `gatsby-remark-images`
        ]
      }
    },
    'gatsby-plugin-emotion',
    'gatsby-plugin-react-helmet',
    'gatsby-theme-style-guide',
    'gatsby-transformer-json',
    `gatsby-plugin-theme-ui`
  ]
})
