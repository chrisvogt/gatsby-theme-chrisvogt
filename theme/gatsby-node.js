const path = require('path')
const startCase = require('lodash/startCase')

const getNodePath = node => {
  let nodePath = '/'

  const category = node.fields.category
  if (category) {
    nodePath += `${category}/`
  }

  const slug = node.frontmatter.slug
  if (slug) {
    nodePath += slug
  }

  return nodePath
}

exports.createPages = async ({ actions, graphql, reporter }) => {
  actions.createPage({
    component: path.resolve('../theme/src/templates/home.js'),
    path: '/'
  })

  const result = await graphql(
    `
      {
        allMdx {
          edges {
            node {
              id
              fields {
                category
                slug
                type
              }
              frontmatter {
                slug
              }
              internal {
                contentFilePath
              }
            }
          }
        }
      }
    `
  )

  if (result.errors) {
    reporter.panic('error loading content', result.errors)
    return
  }

  result.data.allMdx.edges.forEach(({ node }) => {
    const nodePath = getNodePath(node)
    const template =
      node.fields.type === 'media'
        ? path.resolve('../theme/src/templates/media.js')
        : path.resolve('../theme/src/templates/post.js')

    actions.createPage({
      path: nodePath ? nodePath : '/',
      component: `${template}?__contentFilePath=${node.internal.contentFilePath}`,
      context: {
        id: node.id
      }
    })
  })
}

exports.onCreateNode = ({ node, getNode, actions, reporter }) => {
  const { createNodeField } = actions

  if (node.internal.type === `Mdx`) {
    const parent = getNode(node.parent)
    const title = node.frontmatter.title || startCase(parent.name)

    const slug = node.frontmatter.slug
    if (!slug) {
      reporter.panic(
        `Can not create node with title: ${title} there is no relative path or frontmatter to set the "slug" field`
      )
      return
    }

    if (slug === 'index') {
      slug = ''
    }

    const category = node.frontmatter.category
    if (category) {
      createNodeField({
        name: `category`,
        node,
        value: category
      })
    }

    const type = node.frontmatter.type
    if (type) {
      createNodeField({
        name: `type`,
        node,
        value: type
      })
    }

    createNodeField({
      name: `slug`,
      node,
      value: slug
    })

    createNodeField({
      name: 'id',
      node,
      value: node.id
    })

    if (node.internal.type === `Mdx`) {
      createNodeField({
        name: 'path',
        node,
        value: getNodePath(node)
      })
    }

    createNodeField({
      name: 'title',
      node,
      value: title
    })
  }
}
