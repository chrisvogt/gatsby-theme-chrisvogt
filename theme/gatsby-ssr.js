export { default as wrapRootElement } from './wrapRootElement'

export const onRenderBody = ({ setHtmlAttributes }) => {
  setHtmlAttributes({ lang: 'en' })
}
