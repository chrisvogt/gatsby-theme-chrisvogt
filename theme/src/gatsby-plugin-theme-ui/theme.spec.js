import theme, { floatOnHover } from './theme'
import { tailwind } from '@theme-ui/presets'
import { merge } from 'theme-ui'

jest.mock('theme-ui', () => ({
  merge: jest.fn((...args) => Object.assign({}, ...args))
}))

describe('Theme Configuration', () => {
  describe('a snapshot of the configuration', () => {
    it('matches the snapshot', () => {
      expect(theme).toMatchSnapshot()
    })
  })

  describe('general configurations', () => {
    it('merges with Tailwind preset', () => {
      const mergedTheme = merge(tailwind, theme)
      expect(mergedTheme).toBeTruthy()
    })

    it('defaults the color mode to dark', () => {
      expect(theme.config.useColorSchemeMediaQuery).toBe(true)
      expect(theme.config.useCustomProperties).toBe(true)
      expect(theme.config.useLocalStorage).toBe(true)
    })
  })

  describe('font definitions', () => {
    it('defines sans-serif fonts correctly', () => {
      expect(theme.fonts.sans).toContain('sans-serif')
    })

    it('defines serif fonts correctly', () => {
      expect(theme.fonts.serif).toContain('Iowan Old Style')
    })

    it('defines monospace fonts correctly', () => {
      expect(theme.fonts.monospace).toContain('Menlo')
    })
  })

  describe('GradientBanner styles', () => {
    it('defines the GradientBanner styles', () => {
      const banner = theme.styles.GradientBanner
      expect(banner).toHaveProperty('maxWidth', '100%')
      expect(banner).toHaveProperty('height', '340px')
      expect(banner).toHaveProperty('backgroundImage')
      expect(banner).toHaveProperty('animation', 'highlight 3s infinite alternate')
      expect(banner).toHaveProperty('backgroundPosition', '0 0, 0 100%')
      expect(banner).toHaveProperty('backgroundOrigin', 'padding-box, border-box')
      expect(banner).toHaveProperty('backgroundRepeat', 'no-repeat')
      expect(banner).toHaveProperty('backgroundSize', '100% 100%, 100% 200%')
    })

    it('verifies @keyframes highlight definition', () => {
      const highlight = theme.styles.GradientBanner['@keyframes highlight']
      expect(highlight).toBeTruthy()
      expect(highlight['100%']).toHaveProperty('backgroundPosition', '0 0, 0 0')
    })
  })

  describe('PostCard styles', () => {
    it('applies PostCard base styles', () => {
      const postCard = theme.cards.PostCard
      expect(postCard).toHaveProperty('display', 'flex')
      expect(postCard).toHaveProperty('flexDirection', 'column')
      expect(postCard).toHaveProperty('backgroundColor', 'var(--theme-ui-colors-panel-background)')
    })

    it('integrates floatOnHover into PostCard', () => {
      const postCard = theme.cards.PostCard
      expect(postCard).toMatchObject(floatOnHover)
    })

    it('integrates glassmorphismPanel into PostCard', () => {
      const postCard = theme.cards.PostCard
      expect(postCard).toHaveProperty('borderRadius', '10px')
      expect(postCard).toHaveProperty('backdropFilter', 'blur(10px)')
    })

    it('defines PostCard hover styles for .read-more-icon', () => {
      const postCard = theme.cards.PostCard
      expect(postCard['&:hover .read-more-icon']).toHaveProperty('opacity', 1)
      expect(postCard['&:hover .read-more-icon']).toHaveProperty('paddingLeft', '8px')
    })
  })

  describe('Card styles and variants', () => {
    it('defines primary card styles', () => {
      const primaryCard = theme.cards.primary
      expect(primaryCard).toHaveProperty('borderRadius', 'card')
      expect(primaryCard).toHaveProperty('boxShadow', 'default')
    })

    it('defines actionCard with dynamic borderLeft', () => {
      const actionCard = theme.cards.actionCard
      expect(actionCard.borderLeft(theme)).toBe(`2px solid ${theme.colors.primary}`)
      expect(actionCard.a).toHaveProperty(':hover', 'pointer')
    })

    it('tests metricCard nested span styles', () => {
      const metricCard = theme.cards.metricCard
      expect(metricCard.span).toHaveProperty('fontFamily', 'heading')
      expect(metricCard.span).toHaveProperty('fontWeight', 'bold')
      expect(metricCard.span).toHaveProperty('padding', 2)
    })

    it('defines UserProfile card styles dynamically', () => {
      const userProfile = theme.cards.UserProfile
      expect(userProfile.padding(theme)).toEqual(['none', `0 ${theme.space[3]} 0 0`])
      expect(userProfile).toHaveProperty('border', 'none')
      expect(userProfile).toHaveProperty('background', 'none')
    })

    it('tests StatusCardDark styles', () => {
      const statusCardDark = theme.cards.StatusCardDark
      expect(statusCardDark).toHaveProperty('backgroundColor', '#1e2530')
    })
  })

  describe('color mode configurations', () => {
    it('defines light mode background colors', () => {
      expect(theme.colors.background).toBe('#fdf8f5')
      expect(theme.colors['panel-background']).toContain('rgba(255, 255, 255, 0.35)')
    })

    it('defines dark mode background colors', () => {
      const darkMode = theme.colors.modes.dark
      expect(darkMode).toHaveProperty('background', '#1e1e2f')
      expect(darkMode).toHaveProperty('primary', '#1E90FF')
      expect(darkMode).toHaveProperty('text', '#fff')
    })
  })

  describe('responsive layout', () => {
    it('defines container styles', () => {
      const containerStyles = theme.layout.container
      expect(containerStyles).toHaveProperty('maxWidth', ['', '98%', '', '', '1440px'])
      expect(containerStyles).toHaveProperty('py', [2, 3])
      expect(containerStyles).toHaveProperty('px', [3, 4])
    })
  })

  describe('button and badge variants', () => {
    it('defines button variants', () => {
      expect(theme.buttons.primary).toHaveProperty('bg', 'primary')
      expect(theme.buttons.secondary).toHaveProperty('bg', 'secondary')
      expect(theme.buttons.gray).toHaveProperty('bg', 'gray')
    })

    it('defines badge variants', () => {
      expect(theme.badges.primary).toHaveProperty('bg', 'primary')
      expect(theme.badges.outline).toHaveProperty('boxShadow', 'inset 0 0 0 1px')
    })
  })

  describe('GitHubCardFooter styles', () => {
    it('defines display and justify content', () => {
      const footerStyles = theme.styles.GitHubCardFooter
      expect(footerStyles).toHaveProperty('display', 'flex')
      expect(footerStyles).toHaveProperty('justifyContent', 'space-between')
    })
  })

  describe('VideoWrapper styles', () => {
    it('defines wrapper and iframe styles', () => {
      const videoWrapperStyles = theme.styles.VideoWrapper
      expect(videoWrapperStyles).toHaveProperty('position', 'relative')
      expect(videoWrapperStyles).toHaveProperty('paddingBottom', '56.25%')
      expect(videoWrapperStyles.iframe).toHaveProperty('position', 'absolute')
    })
  })
})
