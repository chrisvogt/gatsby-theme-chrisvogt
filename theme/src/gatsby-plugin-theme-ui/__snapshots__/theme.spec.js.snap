// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Configuration a snapshot of the configuration matches the snapshot 1`] = `
{
  "badges": {
    "outline": {
      "bg": "transparent",
      "boxShadow": "inset 0 0 0 1px",
      "color": "primary",
      "fontSize": 1,
    },
    "primary": {
      "bg": "primary",
      "color": "background",
    },
  },
  "borderWidths": {
    "0": "0px",
    "2": "2px",
    "4": "4px",
    "8": "8px",
    "default": "1px",
  },
  "breakpoints": [
    "640px",
    "768px",
    "1024px",
    "1280px",
    "1536px",
  ],
  "buttons": {
    "gray": {
      "bg": "gray",
      "color": "background",
    },
    "primary": {
      "bg": "primary",
      "color": "background",
    },
    "secondary": {
      "bg": "secondary",
      "color": "background",
    },
  },
  "cards": {
    "PostCard": {
      "&:hover .read-more-icon": {
        "opacity": 1,
        "paddingLeft": "8px",
      },
      "&:hover, &:focus": {
        "boxShadow": "lg",
        "transform": "scale(1.015)",
      },
      ".card-media": {
        "height": "100%",
        "mb": 2,
        "overflow": "hidden",
      },
      ".read-more-icon": {
        "display": "inline",
        "opacity": 0,
        "paddingLeft": 0,
        "transition": "all 250ms ease-in",
      },
      "WebkitBackdropFilter": "blur(10px)",
      "backdropFilter": "blur(10px)",
      "background": "var(--theme-ui-colors-panel-background)",
      "backgroundColor": "var(--theme-ui-colors-panel-background)",
      "border": "1px solid rgba(255, 255, 255, 0.15)",
      "borderRadius": "10px",
      "boxShadow": "0 4px 6px rgba(0, 0, 0, 0.1)",
      "color": "var(--theme-ui-colors-panel-text)",
      "display": "flex",
      "flexDirection": "column",
      "flexGrow": 1,
      "fontSize": [
        1,
        2,
      ],
      "height": "100%",
      "padding": 3,
      "textDecoration": "none",
      "transition": "all 200ms ease-in-out",
    },
    "StatusCardDark": {
      "backgroundColor": "#1e2530",
    },
    "UserProfile": {
      "background": "none",
      "border": "none",
      "color": "white",
      "padding": [Function],
    },
    "UserProfileDark": {
      "background": "var(--theme-ui-colors-panel-background)",
      "backgroundColor": "none",
      "borderBottom": "none",
      "borderRadius": "card",
      "boxShadow": "none",
      "color": "white",
      "flexGrow": 1,
      "fontSize": [
        1,
        2,
      ],
      "padding": [Function],
      "textDecoration": "none",
    },
    "actionCard": {
      "&:hover, &:focus": {
        "boxShadow": "lg",
        "transform": "scale(1.015)",
      },
      "WebkitBackdropFilter": "blur(10px)",
      "a": {
        ":hover": "pointer",
      },
      "backdropFilter": "blur(10px)",
      "background": "var(--theme-ui-colors-panel-background)",
      "border": "1px solid rgba(255, 255, 255, 0.15)",
      "borderLeft": [Function],
      "borderRadius": "10px",
      "boxShadow": "0 4px 6px rgba(0, 0, 0, 0.1)",
      "color": "var(--theme-ui-colors-panel-text)",
      "flexGrow": 1,
      "fontSize": [
        1,
        2,
      ],
      "padding": 3,
      "textDecoration": "none",
      "transition": "all 200ms ease-in-out",
    },
    "metricCard": {
      "background": "var(--theme-ui-colors-panel-background)",
      "backgroundColor": "var(--theme-ui-colors-panel-background)",
      "borderRadius": "card",
      "boxShadow": "none",
      "color": "var(--theme-ui-colors-panel-text)",
      "flexGrow": 1,
      "fontSize": [
        1,
        2,
      ],
      "padding": 3,
      "span": {
        "fontFamily": "heading",
        "fontWeight": "bold",
        "padding": 2,
      },
      "textDecoration": "none",
    },
    "primary": {
      "background": "var(--theme-ui-colors-panel-background)",
      "borderRadius": "card",
      "boxShadow": "default",
      "color": "var(--theme-ui-colors-panel-text)",
      "flexGrow": 1,
      "fontSize": [
        1,
        2,
      ],
      "padding": 3,
      "textDecoration": "none",
    },
  },
  "colors": {
    "accent": "deeppink",
    "background": "#fdf8f5",
    "modes": {
      "dark": {
        "background": "#1e1e2f",
        "panel-background": "rgba(1, 1, 1, 0.30)",
        "panel-divider": [Function],
        "panel-highlight": [Function],
        "primary": "#1E90FF",
        "text": "#fff",
        "textMuted": "#d8d8d8",
      },
    },
    "panel-background": "rgba(255, 255, 255, 0.35)",
    "panel-divider": [Function],
    "panel-highlight": [Function],
    "primary": "#422EA3",
    "secondary": "#711E9B",
    "secondaryGradient": "linear-gradient(45deg, #4527a0 0%, #711e9b 100%)",
    "tableText": "#111",
    "text": "#111",
    "textMuted": "#333",
  },
  "config": {
    "useColorSchemeMediaQuery": "initial",
    "useCustomProperties": true,
    "useLocalStorage": true,
  },
  "fontSizes": [
    ".875rem",
    "1rem",
    "1.25rem",
    "1.375rem",
    "1.5rem",
    "1.875rem",
    "2.25rem",
    "3rem",
    "4rem",
    "4.5rem",
  ],
  "fontWeights": {
    "black": 900,
    "body": 400,
    "bold": 700,
    "extrabold": 800,
    "hairline": 100,
    "heading": 700,
    "light": 300,
    "medium": 500,
    "normal": 400,
    "semibold": 600,
    "thin": 200,
  },
  "fonts": {
    "body": "Crimson Text, Georgia, Iowan Old Style, Apple Garamond, Baskerville, Times New Roman, Droid Serif, Times, Source Serif Pro, serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol",
    "heading": "-apple-system, BlinkMacSystemFont, avenir next, avenir, helvetica neue, helvetica, Ubuntu, roboto, noto, segoe ui, arial, sans-serif",
    "monospace": "Menlo, Consolas, Monaco, Liberation Mono, Lucida Console, monospace",
    "sans": "-apple-system, BlinkMacSystemFont, avenir next, avenir, helvetica neue, helvetica, Ubuntu, roboto, noto, segoe ui, arial, sans-serif",
    "serif": "Crimson Text, Georgia, Iowan Old Style, Apple Garamond, Baskerville, Times New Roman, Droid Serif, Times, Source Serif Pro, serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol",
  },
  "global": {
    "@keyframes wobble": {
      "0%, 100%": {
        "transform": "rotate(0deg)",
      },
      "15%": {
        "transform": "rotate(-15deg)",
      },
      "30%": {
        "transform": "rotate(10deg)",
      },
      "45%": {
        "transform": "rotate(-10deg)",
      },
      "60%": {
        "transform": "rotate(5deg)",
      },
      "75%": {
        "transform": "rotate(-5deg)",
      },
    },
    "@media (prefers-reduced-motion: reduce)": {
      ".emoji": {
        "animation": "none !important",
      },
    },
  },
  "inputs": {
    "inline": {
      "&:focus": {
        "backgroundColor": "white",
        "borderColor": "primary",
        "outline": "none",
      },
      "appearance": "none",
      "backgroundColor": "gray.2",
      "borderColor": "gray.2",
      "borderRadius": "default",
      "borderStyle": "solid",
      "borderWidth": "2px",
      "color": "gray.7",
      "fontSize": "100%",
      "lineHeight": "tight",
      "px": 3,
      "py": 2,
    },
    "shadow": {
      "&:focus": {
        "boxShadow": "outline",
        "outline": "none",
      },
      "appearance": "none",
      "border": "none",
      "borderRadius": "default",
      "boxShadow": "default",
      "color": "gray.7",
      "fontSize": "100%",
      "lineHeight": "tight",
      "px": 3,
      "py": 2,
    },
    "underline": {
      "&:focus": {
        "backgroundColor": "white",
        "borderColor": "primary",
        "outline": "none",
      },
      "appearance": "none",
      "backgroundColor": "transparent",
      "border": "none",
      "borderBottomColor": "primary",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "2px",
      "borderRadius": "0px",
      "color": "gray.7",
      "fontSize": "100%",
      "lineHeight": "tight",
      "px": 3,
      "py": 2,
    },
  },
  "layout": {
    "container": {
      "maxWidth": [
        "",
        "98%",
        "",
        "",
        "1440px",
      ],
      "px": [
        3,
        4,
      ],
      "py": [
        2,
        3,
      ],
    },
  },
  "letterSpacings": {
    "normal": "0em",
    "tight": "-0.025em",
    "tighter": "-0.05em",
    "wide": "0.025em",
    "wider": "0.05em",
    "widest": "0.1em",
  },
  "lineHeights": {
    "10": "2.5rem",
    "3": ".75rem",
    "4": "1rem",
    "5": "1.25rem",
    "6": "1.5rem",
    "7": "1.75rem",
    "8": "2rem",
    "9": "2.25rem",
    "body": "1.625",
    "heading": "1.25",
    "loose": "2",
    "none": "1",
    "normal": "1.5",
    "relaxed": "1.625",
    "snug": "1.375",
    "tight": "1.25",
  },
  "links": {
    "homeNavigation": {
      "&:hover, &:focus": {
        "backgroundColor": "var(--theme-ui-colors-panel-background)",
      },
      "&:not(:last-of-type)": {
        "borderBottom": "var(--theme-ui-colors-panel-divider)",
      },
      "color": "primary",
      "display": "block",
      "py": 2,
      "textDecoration": "none",
    },
  },
  "radii": {
    "card": "8px",
    "default": "4px",
  },
  "shadows": {
    "2xl": "0 25px 50px -12px rgb(0 0 0 / 0.25)",
    "default": "0 1px 3px 0 rgb(0 0 0 / 0.1), 0 1px 2px -1px rgb(0 0 0 / 0.1)",
    "inner": "inset 0 2px 4px 0 rgb(0 0 0 / 0.05)",
    "lg": "0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1)",
    "md": "0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1)",
    "sm": "0 1px 2px 0 rgb(0 0 0 / 0.05)",
    "xl": "0 20px 25px -5px rgb(0 0 0 / 0.1), 0 8px 10px -6px rgb(0 0 0 / 0.1)",
  },
  "sizes": {
    "0": "0",
    "1": "0.25rem",
    "1/12": "8.333333%",
    "1/2": "50%",
    "1/3": "33.333333%",
    "1/4": "25%",
    "1/5": "20%",
    "1/6": "16.666667%",
    "10": "2.5rem",
    "10/12": "83.333333%",
    "11/12": "91.666667%",
    "12": "3rem",
    "16": "4rem",
    "2": "0.5rem",
    "2/12": "16.666667%",
    "2/3": "66.666667%",
    "2/4": "50%",
    "2/5": "40%",
    "2/6": "33.333333%",
    "20": "5rem",
    "24": "6rem",
    "2xl": "42rem",
    "3": "0.75rem",
    "3/12": "25%",
    "3/4": "75%",
    "3/5": "60%",
    "3/6": "50%",
    "32": "8rem",
    "3xl": "48rem",
    "4": "1rem",
    "4/12": "33.333333%",
    "4/5": "80%",
    "4/6": "66.666667%",
    "40": "10rem",
    "48": "12rem",
    "4xl": "56rem",
    "5": "1.25rem",
    "5/12": "41.666667%",
    "5/6": "83.333333%",
    "56": "14rem",
    "5xl": "64rem",
    "6": "1.5rem",
    "6/12": "50%",
    "64": "16rem",
    "6xl": "72rem",
    "7/12": "58.333333%",
    "7xl": "80rem",
    "8": "2rem",
    "8/12": "66.666667%",
    "9/12": "75%",
    "full": "100%",
    "lg": "32rem",
    "md": "28rem",
    "prose": "65ch",
    "px": "1px",
    "screenHeight": "100vh",
    "screenWidth": "100vw",
    "sm": "24rem",
    "xl": "36rem",
    "xs": "20rem",
  },
  "space": [
    "0",
    "0.25rem",
    "0.5rem",
    "1rem",
    "2rem",
    "4rem",
    "8rem",
    "16rem",
    "32rem",
  ],
  "styles": {
    ".footnotes": {
      "fontSize": [Function],
    },
    ".text-center": {
      "textAlign": "center",
    },
    "Book": {
      "&:hover, &:focus": {
        "filter": [Function],
        "transform": "scale(1.01)",
        "transition": "all .35s ease-in-out",
      },
      "filter": [Function],
    },
    "Container": {
      "px": [
        3,
        4,
      ],
      "py": [
        2,
        3,
      ],
    },
    "GitHubCardFooter": {
      "display": "flex",
      "justifyContent": "space-between",
      "mt": 2,
    },
    "GradientBanner": {
      "@keyframes highlight": {
        "100%": {
          "backgroundPosition": "0 0, 0 0",
        },
      },
      "alignItems": "center",
      "animation": "highlight 3s infinite alternate",
      "backgroundImage": "
  linear-gradient(white, white),
  linear-gradient(180deg, cornflowerblue, purple 50%, cornflowerblue)",
      "backgroundOrigin": "padding-box, border-box",
      "backgroundPosition": "0 0, 0 100%",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "100% 100%, 100% 200%",
      "border": "20px solid transparent",
      "boxSizing": "border-box",
      "color": "black",
      "display": "flex",
      "height": "340px",
      "justifyContent": "center",
      "maxWidth": "100%",
      "padding": "1rem",
    },
    "GradientBannerDark": {
      "@keyframes highlight": {
        "100%": {
          "backgroundPosition": "0 0, 0 0",
        },
      },
      "alignItems": "center",
      "animation": "highlight 3s infinite alternate",
      "backgroundImage": "
        linear-gradient(#252e3c, #252e3c),
        linear-gradient(270deg, #00D7B9, #B95DD7 50%, #FFB367 100%);",
      "backgroundOrigin": "padding-box, border-box",
      "backgroundPosition": "0 0, 0 100%",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "100% 100%, 100% 200%",
      "border": "20px solid transparent",
      "boxSizing": "border-box",
      "color": "light",
      "display": "flex",
      "height": "340px",
      "justifyContent": "center",
      "maxWidth": "100%",
      "padding": "1rem",
    },
    "Header": {
      "alignItems": "center",
      "color": "text",
      "display": "block",
      "transition": "all 0.3s ease-in-out",
      "width": "100%",
    },
    "InstagramItem": {
      "&:hover, &:focus": {
        "boxShadow": "lg",
        "transform": "scale(1.015)",
      },
      "background": "none",
      "border": "none",
      "borderRadius": "8px",
      "boxShadow": "md",
      "cursor": "pointer",
      "overflow": "hidden",
      "p": 0,
      "transition": "all 200ms ease-in-out",
    },
    "IntroExperienceSlide": {
      "&.active-slide": {
        "display": "block",
        "height": "auto",
        "opacity": 1,
        "visibility": "initial",
      },
      "display": "flex",
      "height": 0,
      "opacity": 0,
      "visibility": "hidden",
    },
    "PageFooter": {
      "a": {
        "color": "text",
      },
      "color": "text",
      "display": "block",
      "width": "100%",
      "zIndex": "10",
    },
    "TopNavigation": {
      "color": "white",
    },
    "VideoWrapper": {
      "height": 0,
      "iframe": {
        "border": 0,
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      },
      "paddingBottom": "56.25%",
      "paddingTop": "25px",
      "position": "relative",
    },
    "WidgetFooter": {
      "color": "text",
      "fontFamily": "heading",
      "textAlign": [
        "center",
        "right",
      ],
    },
    "outlined": {
      "border": "4px solid #efefef",
    },
    "p": {
      "fontSize": [
        2,
        3,
      ],
    },
    "root": {
      "color": "text",
      "display": "flex",
      "flexDirection": "column",
      "fontFamily": "body",
      "fontWeight": "body",
      "lineHeight": "body",
      "minHeight": "100vh",
    },
    "table": {
      "backgroundColor": "light",
      "borderCollapse": "collapse",
      "borderRadius": "10px",
      "borderSpacing": 0,
      "boxShadow": "0 0 10px rgba(0, 0, 0, 0.1)",
      "color": "tableText",
      "marginBottom": "1.5rem",
      "overflow": "hidden",
      "tbody tr:nth-of-type(odd)": {
        "backgroundColor": "#fafafa",
      },
      "th": {
        "backgroundColor": "#f4f4f9",
        "borderColor": "muted",
        "borderTop": "2px solid",
        "fontWeight": "bold",
        "letterSpacing": "0.05em",
        "textTransform": "uppercase",
      },
      "th, td": {
        "borderBottom": "1px solid",
        "borderColor": "muted",
        "fontSize": "1.125em",
        "padding": "12px 15px",
        "textAlign": "left",
      },
      "width": "100%",
    },
    "text": {
      "inverse": {
        "color": "muted",
      },
      "title": {
        "fontFamily": "fonts.sans",
      },
    },
    "ul": {
      "fontSize": [
        2,
        3,
      ],
    },
  },
  "text": {
    "title": {
      "color": "primary",
      "fontFamily": "heading",
      "fontSize": [
        1,
        2,
      ],
      "fontWeight": "550",
      "textTransform": "uppercase",
    },
  },
  "transforms": {
    "rotate": {
      "-180": "-180deg",
      "-45": "-45deg",
      "-90": "-90deg",
      "0": "0",
      "180": "180deg",
      "45": "45deg",
      "90": "90deg",
    },
    "scale": {
      "0": "0",
      "100": "1",
      "105": "1.05",
      "110": "1.1",
      "125": "1.25",
      "150": "1.5",
      "50": ".5",
      "75": ".75",
      "90": ".9",
      "95": ".95",
    },
    "skew": {
      "-12": "-12deg",
      "-3": "-3deg",
      "-6": "-6deg",
      "0": "0",
      "12": "12deg",
      "3": "3deg",
      "6": "6deg",
    },
    "transformOrigin": {
      "bottom": "bottom",
      "bottom-left": "bottom left",
      "bottom-right": "bottom right",
      "center": "center",
      "left": "left",
      "right": "right",
      "top": "top",
      "top-left": "top left",
      "top-right": "top right",
    },
    "translate": {
      "-1/2": "-50%",
      "-full": "-100%",
      "0": "0",
      "1": "0.25rem",
      "1/2": "50%",
      "10": "2.5rem",
      "12": "3rem",
      "16": "4rem",
      "2": "0.5rem",
      "20": "5rem",
      "24": "6rem",
      "3": "0.75rem",
      "32": "8rem",
      "4": "1rem",
      "40": "10rem",
      "48": "12rem",
      "5": "1.25rem",
      "56": "14rem",
      "6": "1.5rem",
      "64": "16rem",
      "8": "2rem",
      "full": "100%",
      "px": "1px",
    },
  },
  "transitions": {
    "duration": {
      "100": "100ms",
      "1000": "1000ms",
      "150": "150ms",
      "200": "200ms",
      "300": "300ms",
      "500": "500ms",
      "700": "700ms",
      "75": "75ms",
      "default": "150ms",
    },
    "property": {
      "all": "all",
      "colors": "background-color, border-color, color, fill, stroke",
      "default": "background-color, border-color, color, fill, stroke, opacity, box-shadow, transform",
      "none": "none",
      "opacity": "opacity",
      "shadow": "box-shadow",
      "transform": "transform",
    },
    "timingFunction": {
      "default": "cubic-bezier(0.4, 0, 0.2, 1)",
      "in": "cubic-bezier(0.4, 0, 1, 1)",
      "in-out": "cubic-bezier(0.4, 0, 0.2, 1)",
      "linear": "linear",
      "out": "cubic-bezier(0, 0, 0.2, 1)",
    },
  },
  "variants": {
    "cards": {
      "dark": {
        "backgroundColor": "teal",
      },
    },
  },
  "zIndices": {
    "0": 0,
    "10": 10,
    "20": 20,
    "30": 30,
    "40": 40,
    "50": 50,
    "auto": "auto",
  },
}
`;
